xrmocap/core/evaluation/fourd_association_evaluation.py

```diff
@@ -0,0 +1,440 @@
+            selected_limbs_name (List[List[str]]): The name of selected
+                limbs in evaluation.
+            additional_limbs_names (List[List[str]]):
+                Names at both ends of the limb.
+            dataset (Union[dict, MviewMpersonDataset])
+            associator (Union[dict, MvposeAssociator])
+            dataset_visualization
+                (Union[None, dict, BaseDataVisualization], optional):
+                Defaults to None.
+            pred_kps3d_convention (str, optional): Target convention of
+                keypoints3d, Defaults to 'coco'.
+            eval_kps3d_convention (str, optional): the convention of
+                keypoints3d for evaluation, Defaults to 'campus'.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+
+        self.output_dir = output_dir
+        self.pred_kps3d_convention = pred_kps3d_convention
+        self.eval_kps3d_convention = eval_kps3d_convention
+        self.additional_limbs_names = additional_limbs_names
+        self.selected_limbs_name = selected_limbs_name
+        self.logger = get_logger(logger)
+
+        if isinstance(dataset, dict):
+            dataset['logger'] = self.logger
+            self.dataset = build_dataset(dataset)
+        else:
+            self.dataset = dataset
+
+        if isinstance(associator, dict):
+            associator['logger'] = self.logger
+            self.associator = build_fourd_associator(associator)
+        else:
+            self.associator = associator
+
+        if isinstance(dataset_visualization, dict):
+            dataset_visualization['logger'] = self.logger
+            self.dataset_visualization = build_data_visualization(
+                dataset_visualization)
+        else:
+            self.dataset_visualization = dataset_visualization
+
+    def run(self, overwrite: bool = False):
+        prepare_output_path(
+            output_path=self.output_dir,
+            allowed_suffix='',
+            path_type='dir',
+            overwrite=overwrite,
+            logger=self.logger)
+        n_frame = len(self.dataset)
+        n_kps = get_keypoint_num(convention=self.pred_kps3d_convention)
+        pred_kps3d = np.zeros(shape=(n_frame, 1, n_kps, 4))
+        gt_kps3d = None
+        max_identity = 0
+        end_of_clip_idxs = []
+        identities = []
+        for frame_idx, frame_item in enumerate(tqdm(self.dataset)):
+            _, _, _, _, kps3d, end_of_clip, kw_data = frame_item
+            if end_of_clip:
+                end_of_clip_idxs.append(frame_idx)
+            fisheye_list = self.dataset.fisheye_params[0]
+            # prepare input for associate single frame
+            
+            self.associator.set_cameras(fisheye_list)
+
+            predict_keypoints3d, identities = \
+                self.associator.associate_frame(kw_data)
```

Comment:
It's not explicit enough what variables does this function pass. Shall we pass the specific variables?